### DISCUSSION TITLE
How to dump ETH Channel to pcap file ?

### DISCUSSION THREAD

**Discussion by congming** [atac/pychapter10#41] [Q&A] [answered]
# Write packets from channel 42 to PCAP file
    dst = "ETH_extracted_Chapter10.pcap"
    binary_data = bytearray()
    packets_to_write = []
    for packet in C10(filename):
        if packet.channel_id == 42:
            try:
                # Assuming packet.data contains the raw bytes of the UDP multicast packet
                udp_multicast_packet = packet.Message
  
                # Create a Scapy packet with Ethernet, IP, and UDP layers
                scapy_packet = Ether() / IP() / UDP() / udp_multicast_packet

                # Append the packet to the list of packets to be written
                packets_to_write.append(scapy_packet)
            except Exception as e:
                print(f"Error processing packet: {e}")

**Comment by mcferrill** [comment, marked as answer]
There's a good example in the dump command in [c10-tools](https://github.com/atac/c10-tools/blob/1562de718a92f4c4e0e30b4e6673dd9108f7077c/c10_tools/dump.py#L64) using dpkt. Feel free to just use that implementation if it's useful, too.
